src/consensus/softforks.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2015 Eric Lombrozo
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "consensus/softforks.h"
+
+#include "chain.h"
+#include "consensus/blockruleindex.h"
+#include "consensus/params.h"
+
+using namespace Consensus::SoftForks;
+
+/**
+ * Returns true if there are nRequired or more blocks of minVersion or above
+ * in the last Consensus::Params::nMajorityWindow blocks, starting at pstart and going backwards.
+ */
+static bool IsSuperMajority(int minVersion, const CBlockIndex* pstart, unsigned nRequired, const Consensus::Params& consensusParams)
+{
+    unsigned int nFound = 0;
+    for (int i = 0; i < consensusParams.nMajorityWindow && nFound < nRequired && pstart != NULL; i++)
+    {
+        if (pstart->nVersion >= minVersion)
+            ++nFound;
+        pstart = pstart->pprev;
+    }
+    return (nFound >= nRequired);
+}
+
+VersionStatus Consensus::SoftForks::CheckVersion(const CBlockIndex& blockIndex, const Consensus::VersionBits::BlockRuleIndex& blockRuleIndex, const Consensus::Params& consensusParams, CBlockIndex* pindexPrev)
```

Comment:
Can't this return a boolean and use ```CValidationState& state``` like ```main::ContextualCheckBlockHeader()``` (the functions where some of ```Consensus::SoftForks::CheckVersion()``` comes from) is currently doing?